Doctest for find_closest_cell is flaky

The doctest of `find_closest_cell` has random data:
```py
        Find nearest cell to a point on a sphere, centered on the
        origin.

        >>> import pyvista
        >>> mesh = pyvista.Sphere()
        >>> index = mesh.find_closest_cell([0, 0, 0.5])
        >>> index
        30

        Find the nearest cells to several random points that
        are centered on the origin.

        >>> import numpy as np
        >>> points = 2 * np.random.random((5000, 3)) - 1
        >>> indices = mesh.find_closest_cell(points)
        >>> indices.shape
        (5000,)

        The average position of all the randomly found cell centers should
        be reasonably close to the origin.

        >>> cell_center_mesh = mesh.cell_centers()
        >>> avg_pos = cell_center_mesh.points[indices, :].mean(axis=0)
        >>> np.linalg.norm(avg_pos) < 0.02
        True
```

This can sometimes trip CI when the alignment of the stars is grim:
```
=================================== FAILURES ===================================
___________ [doctest] pyvista.core.dataset.DataSet.find_closest_cell ___________
1917         >>> indices = mesh.find_closest_cell(points)
1918         >>> indices.shape
1919         (5000,)
1920 
1921         The average position of all the randomly found cell centers should
1922         be reasonably close to the origin.
1923 
1924         >>> cell_center_mesh = mesh.cell_centers()
1925         >>> avg_pos = cell_center_mesh.points[indices, :].mean(axis=0)
1926         >>> np.linalg.norm(avg_pos) < 0.02
Expected:
    True
Got:
    False
```

We should think a bit harder about the test, and maybe come up with something reliable or deterministic (preferably without setting a random seed in the test, because that's bad practice that I'd prefer to avoid in our examples).

--- a/pyvista/core/dataset.py
+++ b/pyvista/core/dataset.py
@@ -1901,7 +1901,7 @@ class DataSet(DataSetFilters, DataObject):
         Examples
         --------
         Find nearest cell on a sphere centered on the
-        origin to the point [0.1, 0.2, 0.3].
+        origin to the point ``[0.1, 0.2, 0.3]``.
 
         >>> import pyvista
         >>> mesh = pyvista.Sphere()
@@ -1911,7 +1911,7 @@ class DataSet(DataSetFilters, DataObject):
         591
 
         Make sure that this cell indeed is the closest to
-        (0.1, 0.2, 0.3).
+        ``[0.1, 0.2, 0.3]``.
 
         >>> import numpy as np
         >>> cell_centers = mesh.cell_centers()

--- a/pyvista/core/dataset.py
+++ b/pyvista/core/dataset.py
@@ -1900,31 +1900,33 @@ class DataSet(DataSetFilters, DataObject):
 
         Examples
         --------
-        Find nearest cell to a point on a sphere, centered on the
-        origin.
+        Find nearest cell on a sphere centered on the
+        origin to the point [0.1, 0.2, 0.3].
 
         >>> import pyvista
         >>> mesh = pyvista.Sphere()
-        >>> index = mesh.find_closest_cell([0, 0, 0.5])
+        >>> point = [0.1, 0.2, 0.3]
+        >>> index = mesh.find_closest_cell(point)
         >>> index
-        30
+        591
+
+        Make sure that this cell indeed is the closest to
+        (0.1, 0.2, 0.3).
+
+        >>> import numpy as np
+        >>> cell_centers = mesh.cell_centers()
+        >>> relative_position = cell_centers.points - point
+        >>> distance = np.linalg.norm(relative_position, axis=1)
+        >>> np.argmin(distance)
+        591
 
         Find the nearest cells to several random points that
         are centered on the origin.
 
-        >>> import numpy as np
         >>> points = 2 * np.random.random((5000, 3)) - 1
         >>> indices = mesh.find_closest_cell(points)
         >>> indices.shape
         (5000,)
-
-        The average position of all the randomly found cell centers should
-        be reasonably close to the origin.
-
-        >>> cell_center_mesh = mesh.cell_centers()
-        >>> avg_pos = cell_center_mesh.points[indices, :].mean(axis=0)
-        >>> np.linalg.norm(avg_pos) < 0.02
-        True
 
         """
         if isinstance(point, collections.abc.Sequence):

